.github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1

```diff
@@ -1,11 +1,24 @@
 param ($MaxParallelThreads)
 
-Write-Output "Replacing maxParallelThreads in xunit.runner.json files to $MaxParallelThreads."
+Write-Output "Setting maxParallelThreads in xunit.runner.json files to $MaxParallelThreads."
 
-($configFiles = Get-ChildItem "*/**/xunit.runner.json" -Recurse) | ForEach-Object {
-     (Get-Content $_) |
-        ForEach-Object  {$_ -Replace '"maxParallelThreads":\s*([-\d]*)', "`"maxParallelThreads`": $MaxParallelThreads"} |
-        Set-Content $_
+$parameters = @{
+        # -Filter is the fastest way to find files because its value is passed directly to the file system API.
+        Filter = "xunit.runner.json"
+        Recurse = $true
+        # -FollowSymLink is needed because without it errors will be thrown of type: 
+        # "Get-ChildItem: The system cannot find the path specified."
+        FollowSymlink = $true
+        # Ignore warnings that will be printed due to multiple symlinks pointing to the same physical file.
+        WarningAction = "Ignore"
+}
+
+($configFiles = Get-ChildItem @parameters) | ForEach-Object {
```

Comment:
Are we really going to stand for inline variable assignment? Static code analysis bans it in C#, so probably shouldn't exist in PS either.

Reply:
Well, I didn't like that either, but it's not my code.

I thought the original author did that on purpose, and not knowing that purpose I kept it like that.

But now, with a third pair of eyes, I'm more confident to refactor that.